components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/internal/APIManagerComponentAddRxtConfigsTest.java

```diff
@@ -0,0 +1,141 @@
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.utils.FileUtil;
+
+import java.io.File;
+import java.io.IOException;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ APIManagerComponent.class, ServiceReferenceHolder.class, FileUtil.class })
+public class APIManagerComponentAddRxtConfigsTest {
+    @Rule
+    private TemporaryFolder folder = new TemporaryFolder();
+    private ServiceReferenceHolder serviceReferenceHolder;
+    private RegistryService registryService;
+    private UserRegistry userRegistry;
+    private Resource resource;
+    private ComponentContext componentContext;
+
+    @BeforeClass
+    public static void setupClass() {
+        System.setProperty("carbon.home", "");
+    }
+
+    @Before
+    public void setup() throws IOException {
+        PowerMockito.mockStatic(ServiceReferenceHolder.class);
+        PowerMockito.mockStatic(FileUtil.class);
+        serviceReferenceHolder = Mockito.mock(ServiceReferenceHolder.class);
+        componentContext = Mockito.mock(ComponentContext.class);
+        registryService = Mockito.mock(RegistryService.class);
+        userRegistry = Mockito.mock(UserRegistry.class);
+        resource = Mockito.mock(Resource.class);
+
+        Mockito.when(serviceReferenceHolder.getRegistryService()).thenReturn(registryService);
+        PowerMockito.when(ServiceReferenceHolder.getInstance()).thenReturn(serviceReferenceHolder);
+    }
+
+    @Test
+    public void testShouldAddRxtConfigs() throws Exception {
+        folder.newFile("abc.rxt");
+        folder.newFile("abcd.rxt");
+        Mockito.when(registryService.getRegistry(Mockito.anyString())).thenReturn(userRegistry);
+        Mockito.when(userRegistry.resourceExists(Mockito.anyString())).thenReturn(true).thenReturn(false);
+        Mockito.when(userRegistry.newResource()).thenReturn(resource);
+        Mockito.when(userRegistry.put(Mockito.anyString(), Mockito.any(Resource.class)))
+                .thenThrow(IndexOutOfBoundsException.class);
+        PowerMockito.when(FileUtil.readFileToString(Mockito.anyString())).thenReturn("TEXT");
+        PowerMockito.whenNew(File.class).withArguments(Mockito.anyString()).thenReturn(folder.getRoot());
+        APIManagerComponent apiManagerComponent = new APIManagerComponent();
+
+        try {
+            // Intentionally throwing an exception after the method to stop further execution of activate();
+            apiManagerComponent.activate(componentContext);
+            Assert.fail("IndexOutOfBoundsException is expected");
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Remove this catch block.

Reply:
IndexOutOfBoundsException is the assertion here. It is expected. If this is removed it will be caught by the Exception catch block. Which is an error scenario

Reply:
Shall we add assertTrue here then?